From 9adc0758fc9240005c49e44279f5d3df06b770da Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
Date: Mon, 9 Apr 2018 23:26:58 +0700
Subject: [PATCH 243/433] serial: sh-sci: Fix kernel panic after closing driver

	Commit: <a6b0756821cd4a8b03876c1b3485f79324edab6b>
	From: 	git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-backport.git

	The rx_timer which has been set up is not deleted when driver is closed.
	A timer is set up with a corresponding callback function when DMA
	transfer is started.  After closing SCIF driver, opens it again, IRQs
	are enabled.  At this time, time out happens and callback function,
	rx_timer_fn() is executed.

[kazuya.mizuguchi.ks@renesas.com:rewrote changelog]
Signed-off-by: Shinya Kuribayashi <shinya.kuribayashi.px@renesas.com>
Signed-off-by: Kazuya Mizuguchi <kazuya.mizuguchi.ks@renesas.com>
Signed-off-by: tungvo <tung.vo.eb@rvc.renesas.com>

Signed-off-by: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
---
 drivers/tty/serial/sh-sci.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/tty/serial/sh-sci.c b/drivers/tty/serial/sh-sci.c
index 0343032..f9eae4c 100644
--- a/drivers/tty/serial/sh-sci.c
+++ b/drivers/tty/serial/sh-sci.c
@@ -1485,6 +1485,8 @@ static void sci_free_dma(struct uart_port *port)
 {
 	struct sci_port *s = to_sci_port(port);
 
+	del_timer_sync(&s->rx_timer);
+
 	if (s->chan_tx)
 		sci_tx_dma_release(s, false);
 	if (s->chan_rx)
-- 
2.7.4

